internal/transport/transport_test.go

```diff
@@ -2746,3 +2746,71 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 		t.Errorf("Context timed out")
 	}
 }
+
+// Test that in the event of a graceful client transport shutdown
```

Comment:
I think this test can be rewritten as follows:

```golang
// Tests the scenario where a client transport is closed and writing of the
// GOAWAY frame as part of the close does not complete because of a network
// hang. The test verifies that the client transport is closed without waiting
// for too long.
func (s) TestClientCloseReturnsEarlyWhenGoAwayWriteHangs(t *testing.T) {
	// Override timer for writing GOAWAY to 0 so that the connection write
	// always times out. It is equivalent of real network hang when conn
	// write for goaway doesn't finish in specified deadline
	origGoAwayLoopyTimeout := goAwayLoopyWriterTimeout
	goAwayLoopyWriterTimeout = 0
	defer func() {
		goAwayLoopyWriterTimeout = origGoAwayLoopyTimeout
	}()

	server, ct, cancel := setUp(t, 0, normal)
	defer cancel()
	defer server.stop()

	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
	defer cancel()
	if _, err := ct.NewStream(ctx, &CallHdr{}); err != nil {
		t.Fatalf("Failed to open stream: %v", err)
	}

	ctClosed := make(chan struct{})
	go func() {
		// ct.Close will try to send a GOAWAY frame to the server but conn.Write
		// will time out due to goAwayLoopyWriterTimeout being 0. It is
		// equivalent of a network hang when GOAWAY doesn't finish within
		// specified deadline.
		ct.Close(errors.New("manually closed by client"))
		close(ctClosed)
	}()

	select {
	case <-ctClosed:
	case <-ctx.Done():
		t.Errorf("Timeout waiting for Close() to complete")
	}
}
```

Reply:
Ok, thAnks. I have made the change and updated the PR.